### PR TITLE
Pdf Highlighting functionality included

### PR DESCRIPTION
**Pull request by Scott771**
Add PDF highlighting features to the toolkit. Refactored row association

### CODE REVIEW

File: indico_toolkit/association/association.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+from typing import Union, List, Dict
+from collections import defaultdict
+from indico_toolkit.types import Predictions
+
+
+class Association:
```
Comment (chris-wells-13) [inline indico_toolkit/association/association.py:6]: should we think about using abc to make abstract base classes more explicit?

File: indico_toolkit/association/association.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        return page_map
+    
+    def validate_prediction_formatting(
+        self, predictions: Union[List[dict], Predictions]
+    ) -> List[dict]:
+        if isinstance(predictions, Predictions):
+            predictions = predictions.to_list()
+        return predictions
+
+    def _is_manually_added(self, prediction: dict):
```
Comment (chris-wells-13) [inline indico_toolkit/association/association.py:42]: this function (given that we're passing a mutable by reference) has a side-effect on the prediction.  is there a way to make that explicit...perhaps the function name?

File: indico_toolkit/association/line_items.py [hunk 3]
```diff
@@ -35,31 +36,28 @@ def __init__(
...
-
         """
-        if isinstance(predictions, Predictions):
-            predictions = predictions.to_list()
+        self.predictions = self.validate_prediction_formatting(predictions)
         if line_item_fields is None:
-            print("No line item fields provided. Will treat all predictions as line items")
-            line_item_fields = Predictions.get_extraction_labels_set(predictions)
+            print(
+                "No line item fields provided. Will treat all predictions as line items"
```
Comment (chris-wells-13) [inline indico_toolkit/association/line_items.py:43]: this is assuming that the user's intention - in not passing line_item_fields - is that they have pre-filtered the predictions down to those that need to be grouped?  i think i would prefer to force the user to feed all predictions + line_item_fields list.

File: indico_toolkit/association/line_items.py [hunk 4]
```diff
@@ -123,14 +114,14 @@ def assign_row_number(self):
         Adds a row_number:int key/val pair based on bounding box position and page
 
         Updates:
-        self._line_item_predictions (list of dicts): predictions with row_number added
+        self._mapped_positions (list of dicts): predictions with row_number added
```
Comment (chris-wells-13) [inline indico_toolkit/association/line_items.py:117]: do these get sorted (either before this class or after)?

File: indico_toolkit/pipelines/file_processing.py [hunk 5]
```diff
@@ -57,7 +57,7 @@ def remove_files_if_processed(self, processed_files: Iterable[str]):
         """
         unprocessed_filepaths = []
         for filepath in self.file_paths:
-            if Path(filepath).name not in processed_files:
+            if self.file_name_from_path(filepath) not in processed_files:
                 unprocessed_filepaths.append(filepath)
         print(f"Removing {len(self.file_paths) - len(unprocessed_filepaths)} files from file_paths")
```
Comment (chris-wells-13) [inline indico_toolkit/pipelines/file_processing.py:62]: at some point, we're going to have a lot of print statements.  should Indico Toolkit be generating it's own logs?

File: indico_toolkit/association/line_items.py (hunk 4 above)
Comment (Scott771) [reply indico_toolkit/association/line_items.py:117]: they get sorted when they begin the mapping process in get_bounding_boxes

File: indico_toolkit/pipelines/file_processing.py (hunk 5 above)
Comment (Scott771) [reply indico_toolkit/pipelines/file_processing.py:62]: yeah I'd like to do that- will make it a separate PR later on

File: indico_toolkit/association/line_items.py (hunk 3 above)
Comment (Scott771) [reply indico_toolkit/association/line_items.py:43]: making this change

File: indico_toolkit/association/association.py (hunk 2 above)
Comment (Scott771) [reply indico_toolkit/association/association.py:42]: error key is being added outside of this method now

File: indico_toolkit/association/association.py (hunk 1 above)
Comment (Scott771) [reply indico_toolkit/association/association.py:6]: added this

**Pull request merged**